File: 2022/06/PHP/main.php
<?php

function find_marker ( &$buffer, $marker_size ) {
	$length = strlen( $buffer );
	for ( $pos = $marker_size; $pos <= $length; $pos++ ) {
		$chars = array_unique(
			str_split(
				substr( $buffer, $pos - $marker_size, $marker_size )
			)
		);
		if ( count($chars) == $marker_size )
			return $pos;
	}
	return FALSE;
}

$input = file_get_contents( getenv( 'ADVENT_INPUT' ) );

echo 'PART1: ' . find_marker( $input, 4 ) . "\n";
echo 'PART2: ' . find_marker( $input, 14 ) . "\n";

File: 2022/06/Python/main.py
#!/usr/bin/python3.10

import os

def find_marker(buffer, marker_size):
    for pos in range(marker_size, len(buffer)):
        chars = {}
        for c in buffer[ pos - marker_size : pos ]:
            chars[c] = 1
        if len(chars) == marker_size:
            return pos
    return -1

input = open(os.getenv('ADVENT_INPUT'), "r").read()
print("PART1: %d" % find_marker(input, 4))
print("PART2: %d" % find_marker(input, 14))
